### PR TITLE
REACT-RAILS TWO APLICATION PROJECT 

### PR DESCRIPTION
**Pull request by tchasinga**
Implement a connection between a Ruby on Rails backend and the React frontend.
Understand the pros and cons of different approaches to connecting the Ruby on Rails backend with the React frontend.

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi @tchasinga, 

# Invalid Code Review Request ❌ 

descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**You have submitted a project, that has the wrong Git flow, your pull request should be created from your feature branch to the dev branch and you should add a link to the backend repository on your frontend README** 👇🏽 

> In the README file for the Rails API, add a link to your React app.
> Also, in the README file for the React app, add a link to your Rails API.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @tchasinga,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Add the correct links to the two repositories in the README as required 👍🏽
✅ Used react-router for routing as expected 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] The front-end app is configured correctly but is only showing an empty screen because it cannot fetch greetings as they are not generated due to the seed attribute error the backend is flagging. Kindly work on that 👍🏽
- [ ] Please work on the seed error showing so greetings are generated and shown on the front-end app 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/store/greetingsSlice.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const fetchGreetings = () => async (dispatch) => {
+  try {
+    const res = await fetch('http://localhost:3000/');
+    const data = await res.json();
+    dispatch(setGreetings(data));
+  } catch (error) {
+    console.log(error);
+  }
+};
```
Comment (frank1738) [inline src/store/greetingsSlice.js:31]: - [ ] 🌐 Just a quick heads-up! The URL you're using to fetch a random greeting is set to the root of your backend API. To get that random greeting, you should update the URL to `http://localhost:3000/random_greeting`. Simply append `/random_greeting` to your URL, and you'll be all set to get that greeting! 😄👍

File: src/components/Greetings.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <div>
+      {greetingsCont.length > 0 && (
+        <h2>
+          {
+            greetingsCont[Math.floor(Math.random() * greetingsCont.length)]
+              .greeting
+          }
+        </h2>
+      )}
+    </div>
```
Comment (frank1738) [inline src/components/Greetings.js:23]: - [ ]  💡 After you've fetched the greeting and updated the state, you can directly use greetingsCont.greeting in your greeting component to display the random greeting. The fetched greeting is already random, so there's no need to use `Math.random()` on the data.
- [ ]  👇 Here's an example of how your component could look based on your work:

```
function Greetings() {
  const dispatch = useDispatch();
  const greetingsCont = useSelector((state) => state.greetings.greetings);

  useEffect(() => {
    dispatch(fetchGreetings());
  }, [dispatch]);

  return (
    <div>
      <h2>{greetingsCont.greeting}</h2>
    </div>
  );
}

export default Greetings;


```

- [ ]  This way, you'll simplify your code and still display the wonderful random greetings you've fetched. Keep up the good work! 😊🚀

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @tchasinga , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ The backend API endpoint is generating a random greeting
✅ Used redux for state management
✅ Created a `Greeting` component

#### Keep up the good work! 😊🚀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] 📝 Remember to write your Git commit messages in imperative tense and capitalize them. For example, `Add feature to generate random greeting endpoint` is a great way to follow this convention. 👍

- [ ] 👀 I noticed the front-end repo doesn't quite follow Git flow practice. It looks like you compared the dev branch to the main branch. Remember, with Gitflow, feature branches should be compared to the dev/development branch. Let's stick to these best practices to make our work more appealing to potential recruiters! 🌟😊

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @tchasinga, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Fixed URL ✔️ 
- Removed `Math.random` 👍 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**